compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmMethodGen.java

```diff
@@ -1491,7 +1492,8 @@ void generateLambdaMethod(BIRInstruction ins, ClassWriter cw, String lambdaName)
             mv.visitMethodInsn(INVOKEINTERFACE, OBJECT_VALUE, "call", methodDesc, true);
         } else {
             String jvmClass;
-            String lookupKey = JvmCodeGenUtil.getPackageName(orgName, moduleName, version) + funcName;
+            String encodedFuncName = IdentifierEncoder.encodeIdentifier(funcName);
+            String lookupKey = JvmCodeGenUtil.getPackageName(orgName, moduleName, version) + encodedFuncName;
```

Comment:
@HindujaB Shall we write a test case for this fix? May be as a separate PR if we need to merge this quickly.

Reply:
I'll send a separate PR for the test